core/src/main/scala/org/apache/spark/ContextCleaner.scala

```diff
@@ -293,4 +314,5 @@ private[spark] trait CleanerListener {
   def broadcastCleaned(broadcastId: Long): Unit
   def accumCleaned(accId: Long): Unit
   def checkpointCleaned(rddId: Long): Unit
+  def listenerCleaned(listener: SparkListener): Unit
```

Comment:
looking at other methods in this trait, they all take IDs. Should this method take session id?

Reply:
The session id doesn't have a 1:1 relationship with the listener so I don't think we should use the session id.